notebooks/NIRISS_WFSS_advanced/00_niriss_mast_query_data_setup.ipynb

```diff
@@ -0,0 +1,457 @@
+   "source": [
+    "## Look at the files existing for each of these different levels\n",
+    "files_to_download = []\n",
+    "for index, batch_exposure in enumerate(single_group):\n",
+    "    \n",
+    "    print('*'*50)\n",
+    "    print(f\"Exposure #{index+1} ({batch_exposure['obs_id']})\")\n",
+    "    # pull out the product names from the list to filter\n",
+    "    products = Observations.get_product_list(batch_exposure)\n",
+    "    \n",
+    "    for filetype, query_dict in file_dict.items():\n",
+    "        print('File type:', filetype)\n",
+    "        filtered_products = Observations.filter_products(products,\n",
+    "                                                         productType=query_dict['product_type'],\n",
+    "                                                         productSubGroupDescription=query_dict['productSubGroupDescription'],\n",
+    "                                                         calib_level=query_dict['calib_level'],\n",
+    "                                                         )\n",
+    "        print(filtered_products['productFilename'])\n",
+    "        files_to_download.extend(filtered_products['productFilename'])\n",
+    "        print()\n",
+    "    print('*'*50)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e98933e6",
+   "metadata": {},
+   "source": [
+    "From above, we can see that for each exposure name in the observation list (`obs_table`), there are many associated files in the background that need to be downloaded as well. This is why we need to work in batches to download.\n",
+    "\n",
+    "<a id='downloading'></a>\n",
+    "#### Downloading Data\n",
+    "To actually download the products, provide ``Observations.download_products()`` with a list of the filtered products. \n",
+    "\n",
+    "If the data are proprietary, you may also need to set up your API token. **NEVER** commit your token to a public repository. An alternative is to create a separate configuration file (config_file.yaml) that is readable only to you and has a key: 'mast_token' : _API token_\n",
+    "\n",
+    "To make create a new API token visit to following link: \n",
+    "    https://auth.mast.stsci.edu/token?suggested_name=Astroquery&suggested_scope=mast:exclusive_access"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "id": "fc1897a3",
+   "metadata": {},
+   "source": [
+    "# if needed, create a separate configuration file and replace Observations.download_products() with the following:\n",
+    "\n",
+    "import yaml\n",
+    "\n",
+    "with open(config_file) as f:\n",
+    "    mast_config = yaml.safe_load(f)\n",
+    "        \n",
+    "mysession = Observations(mast_config['mast_token'])\n",
+    "mysession.download_products(filtered_products)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "616ff15c",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "download_dir = 'data'\n",
```

Comment:
Will need to add this folder programmatically if it does not exist.

Reply:
Done! Good catch!